regular_expressions_boundaries_anchors/regular_expressions_boundaries_anchors.md

```diff
@@ -0,0 +1,419 @@
+
+![Using the pattern for the identifier alone results in matching the identifier when it starts a line as well as when it's in the middle of a line,](media/without_anchor.png)<!-- style = "border: 1px solid rgb(var(--color-highlight)); max-width:400px;" -->
+
+However, if [we add the `^` at the start of our pattern](https://regex101.com/r/4HxMlo/1), resulting in `^(?:WT|PBMC)\-\d{5,6}`, we can indicate that our pattern must immediately begin the string:
+
+![Using the pattern that has the anchor followed by the regex for the identifier results in matching the identifier when it starts a line but not matching the identifier when it's in the middle of a line.](media/with_anchor.png)<!-- style = "border: 1px solid rgb(var(--color-highlight)); max-width:400px;" -->
+
+<div class = "important">
+<b style="color: rgb(var(--color-highlight));">Important note</b><br>
+
+To correctly use `^` as the start-of-string token, you must not enclose it in square brackets.  If `^` appears inside square brackets, it will be intepreted as doing something quite different, as you are about to discover.
+
+</div>
+
+### Except: Another Use of the Caret 
+
+On the previous page, we warned you about a potential pitfall:
+
+> To correctly use `^` as the start-of-string token, you must not enclose it in square brackets.  If `^` appears inside square brackets, it will be intepreted as doing something quite different, as you are about to discover.
+
+Let's look at three ways we can use the caret (`^`).
+
+Start-of-string Caret
+-----
+
+If we want to indicate the start of a string, we add the caret at the beginning of our regular expression, being sure to not enclose it in square brackets.  We see an example of this in the regular expression `^(?:WT|PBMC)\-\d{5,6}`, which includes a start of string character followed by a non-capturing group with two alternatives, a hyphen, and then five to six digits.
+
+Except-for Caret
+------
+
+We can also use the caret symbol in an "except for" clause in our regular expression.  For example, what if at a given position in a pattern, any character is possible except for a hash mark (#)?  We can say "except for" by using `^` as the first character within square brackets: `[^#]`.  
+
+We can add multiple characters that should not appear within the square brackets.  If any character except for  #, 5, ., and % are permitted at a given location in a string, we could express that with the regular expression list `[^#5\.%]`. For example, `[ABXYZ]` means "any of these: A, B, X, Y, or Z", while `[^ABXYZ]` means "anything that **isn't** one of these: A, B, X, Y, and Z."  
+
+<div class = "important">
+<b style="color: rgb(var(--color-highlight));">Important note</b><br>
+
+To use `^` as "except for", it has to immediately follow an opening square bracket, and be followed by the characters it is indicating aren't acceptable.  You can't, for example, say "all the digits except for 7" by typing `\d^7` or `[\d^7]`.
```

Comment:
If I wanted any letter except q, would the best way to do that be [A-PR-Za-pr-z]? It might be nice to have an example here showing that you can denote "any letter except q" or something like that even though it is clunky.

Or maybe just continue the example of [0-689] or [0-68-9].